File: leetcode/problems/ts/xor-operation-in-an-array.ts
// by @codeAbinash
// Time : O(n)
// Space : O(1)

function xorOperation(n: number, start: number): number {
   let res = start
   for (let i = 1; i < n; i++)
      start += 2, res ^= start
   return res
}

File: leetcode/problems/c/xor-operation-in-an-array.c
// by @codeAbinash
// Time : O(n)
// Space : O(1)

int xorOperation(int n, int start) {
   int i, res = start;
   for (i = 1; i < n; i++)
      start += 2, res ^= start;
   return res;
}